indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java

```diff
@@ -434,24 +431,26 @@ private void serializeOutIndex(Context context, Bucket bucket, File mergedBase,
       Path finalIndexZipFilePath = new Path(indexBasePath, "index.zip");
       final URI indexOutURI = finalIndexZipFilePath.toUri();
       ImmutableMap<String, Object> loadSpec;
-      if (outputFS instanceof NativeS3FileSystem) {
+      String indexOutScheme = indexOutURI.getScheme();
+
+      if ((indexOutScheme == null) || (indexOutScheme.compareToIgnoreCase("file") == 0)) {
```

Comment:
agree with gian, for us "null" uri scheme means putting segments on the hdfs and not local. we never explicitly set "hdfs://.../" in the segmentOutputPath but just "/foo/bar".